dLux/apertures.py

```diff
@@ -0,0 +1,2641 @@
+import equinox as eqx
+import matplotlib as mpl 
+import matplotlib.pyplot as plt
+import jax.numpy as np
+import jax 
+import dLux
+import abc
+import functools
+from typing import TypeVar 
+
+
+Array = np.ndarray
+Wavefront = dLux.wavefronts.Wavefront
+
+
+__all__ = ["factorial", "CircularAperture", "SquareAperture", 
+    "HexagonalAperture", "RegularPolygonalAperture", 
+    "IrregularPolygonalAperture", "StaticAperture",
+    "AberratedAperture", "StaticAberratedAperture"]
+
+
+two_pi: float = 2. * np.pi
+        
+ 
+def factorial(n : int) -> int:
```

Comment:
This really should be in the utils folder as a standalone function

Reply:
Also I though the whole point was that this works for floats, not just integers? type hints and docs need updating if so

Reply:
Factorial is only defined mathematically for integers. The gamma function is defined for floats. The use of `int` was purposeful, but you are correct, the implementation is agnostic.